test/integration/thruster.cc

```diff
@@ -243,7 +244,8 @@ void ThrusterTest::TestWorld(const std::string &_world,
 
   // Check movement
   if (_namespace != "lowbattery")
-  {
+  { 
+    ASSERT_FALSE(modelPoses.Pos().X().empty())
```

Comment:
This `ASSERT` is a bit funny. We clear the poses of the model just before this. So obviously, this will be false. So we need to check if `modelPoses` is empty inside the loop before accessing it. This can be done by [short-circuit](https://en.wikipedia.org/wiki/Short-circuit_evaluation) evaluation. See the next comment.